pkg/operator/ceph/object/controller.go

```diff
@@ -104,6 +106,22 @@ func (c *ObjectStoreController) onAdd(obj interface{}) {
 	c.acquireOrchestrationLock()
 	defer c.releaseOrchestrationLock()
 
+	if c.clusterSpec.External.Enable {
+		// health check should tell us if the external cluster has been upgraded and display a message
+		externalCephMonVersion, err := client.GetCephMonVersion(c.context, c.namespace)
```

Comment:
This code block to call `GetCephMonVersion()` then `ValidateCephVersionsBetweenLocalAndExternalClusters()` is in four places and looks exactly the same. Can't you combine them to a single method to avoid the duplication? If the unit tests need to not call GetCephMonVersion(), you could break that out into a separate internal method and still avoid this duplication.